mountpoint-s3-fs/tests/mock_s3_tests.rs

```diff
@@ -57,17 +58,23 @@ async fn test_lookup_throttled_mock(head_object_throttled: bool, list_object_thr
     } else {
         (200, &head_object_ok_headers)
     };
-    server.mock(|when, then| {
-        when.method(Method::GET)
-            .path(format!("/{bucket}/"))
-            .query_param("list-type", "2")
-            .query_param("prefix", format!("{key}/"));
-        then.status(list_http_code).body(list_response);
-    });
-    server.mock(|when, then| {
-        when.method(Method::HEAD).path(format!("/{bucket}/{key}"));
-        set_response_headers(then.status(head_object_http_code), head_object_headers);
-    });
+    Mock::given(method("GET"))
+        .and(path(format!("/{bucket}/")))
+        .and(query_param("list-type", "2"))
+        .and(query_param("prefix", format!("{key}/")))
+        .respond_with(ResponseTemplate::new(list_http_code).set_body_string(list_response))
+        .mount(&server)
+        .await;
+
+    let mut head_response = ResponseTemplate::new(head_object_http_code);
+    for (key, value) in head_object_headers {
+        head_response = head_response.insert_header(*key, *value);
+    }
```

Comment:
Couldn't we use `append_headers` here? Or alternatively use a rewritten `set_response_headers`?